### ISSUE TITLE
data store: task outputs are not re-loaded on reload

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#5215]
(requires confirmation) steps to reproduce:

* Write a workflow with a task which produces a custom output.
* Run the workflow, holding a downstream task of the one which produces the custom output (to keep it in the n=1 window).
* Inspect the tasks outputs, they are displayed in the job row in the tree view and can be queried from graphiql.
* Reload the workflow.

Lower priority bug, hence the milestone, can slip if needed.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by dwsutherland** [comment]
Yes, makes sense this is the case.. I assume the job is no longer in the active pool, so it probably needs loaded from the DB?

**Comment by oliver-sanders** [comment]
Makes sense, could we avoid wiping the store on reload or at least just avoid wiping `n=1` completed tasks that remain `n=1` post-reload to avoid needing to go back to the DB?

**Comment by dwsutherland** [comment]
> Makes sense, could we avoid wiping the store on reload or at least just avoid wiping `n=1` completed tasks that remain `n=1` post-reload to avoid needing to go back to the DB?

I had toyed with this idea, but we have to be happy with the fact the outputs (and/or other fields) may have been the subject of the reload.

I think I'd have to move the `self.data[w_id]` to `self.pre_reload[w_id]` temporarily and only target finished/completed and/or non-active parents..

**Comment by oliver-sanders** [comment]
> the outputs (and/or other fields) may have been the subject of the reload.

Ah yeah, that makes sense.
